src/main/java/io/aiven/kafka/auth/audit/UserOperationsActivityAuditor.java

```diff
@@ -16,26 +16,51 @@
 
 package io.aiven.kafka.auth.audit;
 
+import java.util.Objects;
+
+import kafka.network.RequestChannel;
 import kafka.security.auth.Operation;
 import kafka.security.auth.Resource;
 import org.slf4j.Logger;
 
 public class UserOperationsActivityAuditor extends Auditor {
 
     public UserOperationsActivityAuditor() {
+        super();
     }
 
     protected UserOperationsActivityAuditor(final Logger logger) {
         super(logger);
     }
 
     @Override
-    protected UserActivity onUserActivity(final UserActivity userActivity,
-                                          final Operation operation,
-                                          final Resource resource,
-                                          final Boolean hasAccess) {
-        userActivity.addOperation(new UserOperation(operation, resource, hasAccess));
-        return userActivity;
+    protected void addActivity0(final RequestChannel.Session session,
+                                final Operation operation,
+                                final Resource resource,
+                                final boolean hasAccess) {
+        final var grouping = auditorConfig.getAggregationGrouping();
+        final AuditKey auditKey = grouping == AuditorConfig.AggregationGrouping.USER
+                ? new AuditKey(session.principal(), null)
+                : new AuditKey(session.principal(), session.clientAddress());
```

Comment:
Generally, it's better to use `switch` together with enums. `if` with listing all the possible values explicitly works as well. Simple `if-else` and ternary operator are prone to mistake in case a new value is added to the enum.
There are a number of places like this in the code.

Reply:
@willyborankin this is still valid, in some other places of the code.